RecDP/pyrecdp/primitives/operations/text_embedding.py

```diff
@@ -0,0 +1,232 @@
+        self.vector_store = vector_store
+        self.text_embeddings = text_embeddings
+        self.vector_store_args = vector_store_args
+
+    @abstractmethod
+    def persist(self, ds: Union[Dataset, DataFrame]):
+        """persist index store in ray dataset to underlying vector store"""
+
+
+class LangchainVectorStore(VectorStore):
+
+    def __persist_to_faiss(self, ds):
+        import_faiss()
+        text_embeddings: List[Tuple[str, List[float]]] = []
+        if isinstance(ds, Dataset):
+            for row in ds.iter_rows():
+                text_embeddings.append((row[self.text_column], row[self.embeddings_column]))
+        else:
+            for row in ds.collect():
+                text_embeddings.append((row[self.text_column], row[self.embeddings_column]))
+
+        from langchain.vectorstores.faiss import FAISS
+        db = FAISS.from_embeddings(text_embeddings, embedding=self.text_embeddings.underlying_embeddings())
+        if "output_dir" not in self.vector_store_args:
+            raise ValueError(f"You must have `output_dir` option specify for vector store {self.vector_store}")
+
+        index_name = self.vector_store_args["index_name"] if "index_name" in self.vector_store_args else "index"
+        db.save_local(self.vector_store_args["output_dir"], index_name)
+
+    def persist(self, ds):
+        import_langchain()
+        if self.vector_store == "FAISS":
+            self.__persist_to_faiss(ds)
+        else:
+            raise NotImplementedError(f"persist embeddings to vector store '{self.vector_store}' is not supported yet!")
+
+
+def get_vector_store(text_embeddings: TextEmbeddings,
+                     text_column: str = 'text',
+                     embeddings_column: str = 'embedding',
+                     third_party: str = 'langchain',
+                     vector_store: str = 'FAISS',
+                     vector_store_args: Optional[Dict] = None,
+                     ):
+    if third_party is None or third_party == 'langchain':
+        return LangchainVectorStore(
+            text_column=text_column,
+            embeddings_column=embeddings_column,
+            text_embeddings=text_embeddings,
+            vector_store=vector_store,
+            vector_store_args=vector_store_args
+        )
+    else:
+        raise NotImplementedError(f"vector store from {third_party} is not supported yet!")
+
+
+class Embedding:
+    def __init__(self, embeddings: TextEmbeddings, text_column: str = 'text', embeddings_column: str = 'embedding'):
+        # Specify "cuda" to move the model to GPU.
+        self.embeddings = embeddings
+        self.text_column = text_column
+        self.embeddings_column = embeddings_column
+
+    def __call__(self, batch):
+        embeddings = self.embeddings.embed_documents(batch[self.text_column])
+        return {self.text_column: batch[self.text_column], self.embeddings_column: embeddings}
+
+
+class BaseDocumentEmbedding(BaseLLMOperation, ABC):
```

Comment:
Should we rename it to DataIngestion? @yao531441 , since this is not only embedding the text, but also insert into VectorStore.
![image](https://github.com/intel/e2eAIOK/assets/4355494/0a09ed66-824b-4b5e-8ff6-0cdf368bbc91)